cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -228,6 +228,11 @@ func ReadinessProbe(host, path string, port int, scheme v1.URIScheme) *v1.Probe
 	return createHTTPProbe(host, path, port, scheme, 0, 15, 3, 1)
 }
 
+// StartupProbe creates a Probe object with a HTTPGet handler
+func StartupProbe(host, path string, port int, scheme v1.URIScheme) *v1.Probe {
+	return createHTTPProbe(host, path, port, scheme, 15, 15, 30, 10)
```

Comment:
The thing that concerns me here is the full timeout. Ideally I would like something like twice the worst case scenario.
@fabriziopandini do you know what's the worst case scenario for an API server to come alive that the Cluster API folks observed during their tests?
@ereslibre I bet that you also have some practical observations here.

Reply:
in kubeadm we wait for a configurable amount of time for static pods to come up.
`timeoutForControlPlane: 4m0s`

so i guess there is no point of making the startup probe timeout longer than that?

Reply:
Thanks for all your feedbacks!
I prefer to use `timeoutForControlPlane` as the full timeout of startup probe for control-plane static Pods.
In this way, uses can change the full timeout(if they want) by:
```
apiServer:
  timeoutForControlPlane: 5m0s
kind: ClusterConfiguratio
apiVersion: kubeadm.k8s.io/v1beta2
```

Reply:
I like that idea, let's do that.